File: vonage_video_bridge/android/src/main/kotlin/com/example/vonage_video_bridge/VonageVideoPlatformView.kt
package com.example.vonage_video_bridge

import android.content.Context
import android.view.View
import android.widget.FrameLayout
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory
import io.flutter.plugin.common.StandardMessageCodec

class VonageVideoPlatformView(
    context: Context,
    private val controller: VonageVideoController
) : PlatformView {
    private val root = FrameLayout(context)

    init {
        controller.attach(root)
    }

    override fun getView(): View = root

    override fun dispose() {
        controller.detach(root)
    }
}

class VonageVideoPlatformViewFactory(
    private val controller: VonageVideoController
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        return VonageVideoPlatformView(context, controller)
    }
}

File: vonage_video_bridge/ios/Classes/VonageVideoPlatformView.swift
import Flutter
import UIKit

final class VonageVideoPlatformViewFactory: NSObject, FlutterPlatformViewFactory {
  private let controller: VonageVideoController

  init(controller: VonageVideoController) {
    self.controller = controller
    super.init()
  }

  func create(
    withFrame frame: CGRect,
    viewIdentifier viewId: Int64,
    arguments args: Any?
  ) -> FlutterPlatformView {
    return VonageVideoPlatformView(frame: frame, controller: controller)
  }
}

final class VonageVideoPlatformView: NSObject, FlutterPlatformView {
  private let root: UIView
  private let controller: VonageVideoController

  init(frame: CGRect, controller: VonageVideoController) {
    self.root = UIView(frame: frame)
    self.controller = controller
    super.init()
    controller.attach(root)
  }

  func view() -> UIView {
    return root
  }

  deinit {
    controller.detach(root)
  }
}
